x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer_utils/timeseriesexplorer_utils.js

```diff
@@ -240,12 +240,12 @@ export function findNearestChartPointToTime(chartData, time) {
     // grab the current and previous items and compare the time differences
     let foundItem;
     for (let i = 0; i < chartData.length; i++) {
-      const itemTime = chartData[i].date.getTime();
+      const itemTime = chartData[i]?.date?.getTime();
       if (itemTime > time) {
         const item = chartData[i];
         const previousItem = chartData[i - 1];
 
-        const diff1 = Math.abs(time - previousItem.date.getTime());
+        const diff1 = Math.abs(time - previousItem?.date?.getTime());
```

Comment:
This `diff1` will be `NaN` if previousItem is undefined - is that intended?

Reply:
This is fine as it's only used for a boolean check